packages/wonder-blocks-data/components/intercept-data.md

```diff
@@ -0,0 +1,163 @@
+import {InterceptData, Data, RequestHandler} from "@khanacademy/wonder-blocks-data";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+class MyHandler extends RequestHandler {
+    constructor() {
+        super("INTERCEPT_DATA_HANDLER1");
+    }
+
+    fulfillRequest(options) {
+        return Promise.reject(new Error("You should not see this!"));
+    }
+}
+
+const handler = new MyHandler();
+const fulfillRequestInterceptor = function(options) {
+    if (options === "DATA") {
+        return Promise.resolve("INTERCEPTED DATA!");
+    }
+    return null;
+};
+
+<InterceptData handler={handler} fulfillRequest={fulfillRequestInterceptor}>
+    <View>
+        <Body>This received intercepted data!</Body>
+        <Data handler={handler} options={"DATA"}>
+            {({loading, data}) => {
+                if (loading) {
+                    return "If you see this, the example is broken!";
+                }
+
+                return (
+                    <BodyMonospace>{data}</BodyMonospace>
+                );
+            }}
+        </Data>
+    </View>
+</InterceptData>
+```
+
+The `shouldRefreshCache` intercept function has the form:
+
+```js static
+(options: TOptions, cachedEntry: ?$ReadOnly<CacheEntry<TData>>) => ?boolean;
+```
+
+If this method is omitted or returns `null`, the default behavior occurs. This
+means that if `cacheEntry` has a value, that will be used; otherwise, the
+`shouldRefreshCache` method of the handler assigned to the `Data` component
+being intercepted will be invoked.
+
+```jsx
+import {Body, BodyMonospace} from "@khanacademy/wonder-blocks-typography";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {withActionScheduler} from "@khanacademy/wonder-blocks-timing";
+import {InterceptData, Data, RequestHandler} from "@khanacademy/wonder-blocks-data";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Color from "@khanacademy/wonder-blocks-color";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+
+class MyHandler extends RequestHandler {
+    constructor() {
+        super("INTERCEPT_DATA_HANDLER2");
+        let _counter = 0;
+        this.fulfillRequest = function(options) {
+            return Promise.resolve(`DATA ${_counter++}`);
+        }
+    }
+
+    shouldRefreshData(options) {
+        return false;
+    }
+}
+
+const handler = new MyHandler();
+const shouldRefreshCacheInterceptor = function(options) {
+    if (options === "DATA") {
+        return true;
+    }
+    return null;
+};
+
+class SchedulableExample extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            stamp: Date.now(),
+        };
+    }
+
+    componentDidMount() {
+        this.props.schedule.interval(() => this.setState({
+            stamp: Date.now(),
+        }), 1000);
```

Comment:
Having a button would allow devs to see the change better.

Reply:
The intent was not really to demonstrate the change so much as to demonstrate the outcome.

Reply:
i.e. the result is important, less-so the journey.